pkg/engine/engine.go

```diff
@@ -1000,13 +1002,14 @@ func (e *Engine) detectChunk(ctx context.Context, data detectableChunk) {
 		detectBytesPerMatch.Observe(float64(len(matchBytes)))
 
 		ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+		t := time.AfterFunc(time.Second*11, func() {
```

Comment:
__optional nitpick__: It might be worth setting the `time.Second * 10` into a variable and use the reference for `AfterFunc` so that an adjustment to the context timeout doesn't cause the `time.Second * 11` to no longer be accurate.